scripts/train_agents.py

```diff
@@ -12,17 +16,55 @@ def SP(args, pop_force_training):
         'generate': [TeamType.SELF_PLAY],
         'load': get_eval_types_to_load()
     }
-    get_selfplay_agent(args=args,
-                       train_types=args.sp_train_types,
-                       eval_types=args.sp_eval_types,
-                       total_training_timesteps=args.pop_total_training_timesteps,
-                       force_training=pop_force_training,
-                        )
+    _, _ = get_selfplay_agent_w_tms_collection(args=args,
+                                            train_types=args.sp_train_types,
+                                            eval_types=args.sp_eval_types,
+                                            total_training_timesteps=args.pop_total_training_timesteps,
+                                            force_training=pop_force_training)
+
+
+def SP_w_SP_Types(args, pop_force_training:bool, parallel:bool) -> None:
+    '''
+    Set up and run the training for self-play with self-play types
+    Similar to FCP_w_SP_TYPES, this function will first train a population of SP agents, organize them into a teammates_collection
+    based on TeamType, and then select agents from th the teams to SP with a randomly initialized SP agent
+    So the randomly initialized agent will train with itself and one other unseen teammate (e.g. [SP, SP, SP, SP_H] in a 4-chef layout)
+    
+    :param pop_force_training: Boolean that, if true, indicates population should be generated, otherwise load it from file
+    :param parallel: Boolean indicating if parallel envs should be used for training or not
+    '''
+
+    # Set the train and eval types that will be used to organize the first "FCP-like" population
+    # NOTE: When in order to use SELF_PLAY_X later, we need X_FIRST here
+    args.sp_train_types = [TeamType.HIGH_FIRST]
+    args.sp_eval_types = {
+        'generate': [TeamType.HIGH_FIRST],
+        'load': []
+        }
+
+    # Set the train and eval types for the teammates that will be used to play with the SP agent
+    args.sp_w_sp_train_types = [TeamType.SELF_PLAY_HIGH]
+    args.sp_w_sp_eval_types = {
+                            'generate': [TeamType.SELF_PLAY_HIGH],
+                            'load': get_eval_types_to_load()
+                            }
+
+    get_selfplay_agent_trained_w_selfplay_types(
+        args,
+        pop_total_training_timesteps=args.pop_total_training_timesteps,
+        sp_train_types=args.sp_train_types,
+        sp_eval_types=args.sp_eval_types,
+        num_self_play_agents_to_train=args.num_sp_agents_to_train,
+        sp_w_sp_total_training_timesteps=args.sp_w_sp_total_training_timesteps,
+        sp_w_sp_train_types=args.sp_w_sp_train_types,
+        sp_w_sp_eval_types=args.sp_w_sp_eval_types,
+        force_training=pop_force_training,
```

Comment:
You need to pass two different force_training
One is force training the teammates_collection which is called through get_fcp_population(). Basically, your teammates_collection can be loaded from file (code [here](https://github.com/HIRO-group/multiHRI/blob/3b496f7f2ee228698185050b8b4f77943bba8802/scripts/utils/fcp_pop_helper.py#L23)). 
The other is to force train the sp_w_sp_types. (code [here](https://github.com/HIRO-group/multiHRI/blob/3b496f7f2ee228698185050b8b4f77943bba8802/scripts/utils/train_helper.py#L98))